Add Dockerfile and GH actions testing workflow

konst-aa opened the pull request:
cc: @ufo5260987423 

This PR implements a GH actions testing workflow and adds a `Dockerfile`. It aims to resolve #35.

* To achieve this, I also had to change `test.sh`. This PR doesn't modify its behavior, just the exit codes, as I wanted `test.sh` to return 0 if and only if all tests passed.
* The Dockerfile can be modified to build whatever version of Chez Scheme we want, as we override the version of Chez in the Akku Alpine image.

Notes & topics to discuss:
* I still would like to add Nix support, if that's still on the table. I can do it in a different PR, or I can do it in this one and then rewrite the Dockerfile to use the Nix environment. I'm happy with either.

ufo5260987423 commented:
Good job! Though there're still something wrong.

ufo5260987423 commented:
And a suggestion:
The test.sh is mainly used for debug. It means, whether a test is passed or not, the continuation shouldn't be stopped. 

Because here's a case that when programmers just did some changes, it needs all the tests to figure out all possible errors. **Here, the key is not to figure out all possibility, but is a programmer can pick up one and make some detailed analysis.**
A frequent scenario is just fixing the easy one and many others are also fixed. 

So,  I suppose that to keep the test.sh, and to make your change in other file. And with your all workflow files, they shouldn't be in the root directory. You know, not all programmers are that good at the workflow.
How about make another directory named docker?

konst-aa commented:
Oh shoot, I'll look into why the build fails...

I can totally make a directory called Docker or CI, or some other name we like. Though, I don't think it's necessary at the moment as the only "new file" is the Dockerfile:

> The test.sh is mainly used for debug. It means, whether a test is passed or not, the continuation shouldn't be stopped.

Yeah, though when I rewrote `test.sh`, I kept that behavior, I believe. The for loop just runs, no problem. The only thing I changed is that if a test fails, the exit code for the script becomes non-zero, so the GitHub action will fail.

konst-aa commented:
@ufo5260987423, could you try rerunning the action?

ufo5260987423 commented:
It works! Thank you for your contribution!
